data-pipeline/src/pipeline/storage.py

```diff
@@ -48,9 +51,11 @@ def get_blob(container_name, blob_name, encoding=None):
     with blob_client as blob:
         if encoding is None:
             content = blob.download_blob(encoding=encoding).readall()
+            logger.info(f"Downloaded blob: {blob_name}")
```

Comment:
Given we've got the data, I'd suggest we log some integrity-related data along with this:

- size in bytes (should be `len(content)`)
- checksum (`hashlib.sha256(content).hexdigest()` or similar)